### PR TITLE
CRAYSAT-1884: Updating cray-product-catalog to latest version

### PR DESCRIPTION
**Pull request by shivaprasad-metimath**
## Summary and Scope

Updating the cray-product-catalog to latest version 2.3.1

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CRAYSAT-1884](https://jira-pro.it.hpe.com:8443/browse/CRAYSAT-1884)
* Change will also be needed in `<release/5.0>`

## Testing

Tested 

### Tested on:

 Gamora,Rocket

### Test description:

will execute cfs-config-util commands to validate

## Risks and Mitigations

minimal

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by shivaprasad-metimath** [comment]
Tested on Gamora and rocket:

**Rocket: 1.5.x**
ncn-m001:/mnt/developer/shiva/cfs-config-util # cfs-config-util update-configs --base-query role=Management --save-suffix .shiva --product sat --playbook sat-ncn.yml
INFO: Querying CFS configurations for the following NCNs: x3000c0s25b0n0, x3000c0s23b0n0, x3000c0s29b0n0, x3000c0s21b0n0, x3000c0s31b0n0, x3000c0s27b0n0, x3000c0s11b0n0, x3000c0s19b0n0, x3000c0s33b0n0, x3000c0s17b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s25b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s23b0n0
INFO: Found configuration "minimal-management-24.6.0" for component x3000c0s29b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s21b0n0
INFO: Found configuration "minimal-management-24.6.0" for component x3000c0s31b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s27b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s11b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s19b0n0
INFO: Found configuration "minimal-management-24.6.0" for component x3000c0s33b0n0
INFO: Found configuration "management-24.6.0" for component x3000c0s17b0n0
INFO: No ConfigMaps found in namespace services with label type=cray-product-catalog."
INFO: Getting data from ConfigMap services/cray-product-catalog without label type=cray-product-catalog
INFO: Updating existing layer with repo path /vcs/cray/sat-config-management.git and playbook sat-ncn.yml
INFO: Property "name" of layer with repo path /vcs/cray/sat-config-management.git and playbook sat-ncn.yml updated from sat-ncn-2.6.15 to sat-sat-ncn-181682a-20240829T062951
INFO: Successfully saved CFS configuration "management-24.6.0.shiva"
INFO: No layer with repo path /vcs/cray/sat-config-management.git and playbook sat-ncn.yml found.
INFO: Adding a layer with repo path /vcs/cray/sat-config-management.git and playbook sat-ncn.yml to the end.
INFO: Successfully saved CFS configuration "minimal-management-24.6.0.shiva"
INFO: Successfully saved 2 changed CFS configuration(s) to CFS.


**Gamora:1.6.x**
ncn-m001:/mnt/developer/shiva/cfs-config-util # cfs-config-util update-configs --base-query role=Management --save-suffix .shiva --product csm --playbook sat-ncn.yml
INFO: Querying CFS configurations for the following NCNs: x3000c0s1b0n0, x3000c0s13b0n0, x3000c0s9b0n0, x3000c0s7b0n0, x3000c0s15b0n0, x3000c0s28b0n0, x3000c0s17b0n0, x3000c0s3b0n0, x3000c0s5b0n0, x3000c0s11b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s1b0n0
INFO: Found configuration "storage-main-csm-1.6.0-alpha.63" for component x3000c0s13b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s9b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s7b0n0
INFO: Found configuration "storage-main-csm-1.6.0-alpha.63" for component x3000c0s15b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s28b0n0
INFO: Found configuration "storage-main-csm-1.6.0-alpha.63" for component x3000c0s17b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s3b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s5b0n0
INFO: Found configuration "management-main-csm-1.6.0-alpha.63" for component x3000c0s11b0n0
INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook sat-ncn.yml found.
INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook sat-ncn.yml to the end.
INFO: Successfully saved CFS configuration "management-main-csm-1.6.0-alpha.63.shiva"
INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook sat-ncn.yml found.
INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook sat-ncn.yml to the end.
INFO: Successfully saved CFS configuration "storage-main-csm-1.6.0-alpha.63.shiva"
INFO: Successfully saved 2 changed CFS configuration(s) to CFS.
ncn-m001:/mnt/developer/shiva/cfs-config-util #

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+# [5.0.4] - 2024-08-29
```
Comment (annapoorna-s-alt) [inline CHANGELOG.md:28]: Can we use 5.1.0 version here same as per the discussion on  csm-api-client repo

**Review by annapoorna-s-alt — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (shivaprasad-metimath) [reply CHANGELOG.md:28]: Yeah, I had the same thought. But was just sceptical as we are closing to the cut off date and no other changes would go in.

Comment (annapoorna-s-alt) [reply CHANGELOG.md:28]: There will be a good amount of changes will be there after cfs V2 to V3 migration. Good to go with version update

Comment (shivaprasad-metimath) [reply CHANGELOG.md:28]: Sure. will go ahead with 5.1.0 itself

**Pull request merged**